crates/wasi-common/src/error.rs

```diff
@@ -242,6 +248,280 @@ impl Error {
+        wasi::__WASI_ERRNO_AFNOSUPPORT => WSAEAFNOSUPPORT,
+        wasi::__WASI_ERRNO_ALREADY => WSAEALREADY,
+        wasi::__WASI_ERRNO_BADF => WSAEBADF,
+        wasi::__WASI_ERRNO_DESTADDRREQ => WSAEDESTADDRREQ,
+        wasi::__WASI_ERRNO_DQUOT => WSAEDQUOT,
+        wasi::__WASI_ERRNO_FAULT => WSAEFAULT,
+        wasi::__WASI_ERRNO_HOSTUNREACH => WSAEHOSTUNREACH,
+        wasi::__WASI_ERRNO_INPROGRESS => WSAEINPROGRESS,
+        wasi::__WASI_ERRNO_INTR => WSAEINTR,
+        wasi::__WASI_ERRNO_ISCONN => WSAEISCONN,
+        wasi::__WASI_ERRNO_LOOP => WSAELOOP,
+        wasi::__WASI_ERRNO_MFILE => WSAEMFILE,
+        wasi::__WASI_ERRNO_MSGSIZE => WSAEMSGSIZE,
+        wasi::__WASI_ERRNO_NAMETOOLONG => WSAENAMETOOLONG,
+        wasi::__WASI_ERRNO_NETDOWN => WSAENETDOWN,
+        wasi::__WASI_ERRNO_NETRESET => WSAENETRESET,
+        wasi::__WASI_ERRNO_NETUNREACH => WSAENETUNREACH,
+        wasi::__WASI_ERRNO_NOBUFS => WSAENOBUFS,
+        wasi::__WASI_ERRNO_NOPROTOOPT => WSAENOPROTOOPT,
+        wasi::__WASI_ERRNO_NOTEMPTY => WSAENOTEMPTY,
+        wasi::__WASI_ERRNO_NOTSOCK => WSAENOTSOCK,
+        wasi::__WASI_ERRNO_PROTONOSUPPORT => WSAEPROTONOSUPPORT,
+        wasi::__WASI_ERRNO_PROTOTYPE => WSAEPROTOTYPE,
+        wasi::__WASI_ERRNO_STALE => WSAESTALE,
+        wasi::__WASI_ERRNO_IO
+        | wasi::__WASI_ERRNO_ISDIR
+        | wasi::__WASI_ERRNO_2BIG
+        | wasi::__WASI_ERRNO_BADMSG
+        | wasi::__WASI_ERRNO_BUSY
+        | wasi::__WASI_ERRNO_CANCELED
+        | wasi::__WASI_ERRNO_CHILD
+        | wasi::__WASI_ERRNO_DEADLK
+        | wasi::__WASI_ERRNO_DOM
+        | wasi::__WASI_ERRNO_FBIG
+        | wasi::__WASI_ERRNO_IDRM
+        | wasi::__WASI_ERRNO_ILSEQ
+        | wasi::__WASI_ERRNO_MLINK
+        | wasi::__WASI_ERRNO_MULTIHOP
+        | wasi::__WASI_ERRNO_NFILE
+        | wasi::__WASI_ERRNO_NODEV
+        | wasi::__WASI_ERRNO_NOEXEC
+        | wasi::__WASI_ERRNO_NOLCK
+        | wasi::__WASI_ERRNO_NOLINK
+        | wasi::__WASI_ERRNO_NOMEM
+        | wasi::__WASI_ERRNO_NOMSG
+        | wasi::__WASI_ERRNO_NOSPC
+        | wasi::__WASI_ERRNO_NOSYS
+        | wasi::__WASI_ERRNO_NOTDIR
+        | wasi::__WASI_ERRNO_NOTRECOVERABLE
+        | wasi::__WASI_ERRNO_NOTSUP
+        | wasi::__WASI_ERRNO_NOTTY
+        | wasi::__WASI_ERRNO_NXIO
+        | wasi::__WASI_ERRNO_OVERFLOW
+        | wasi::__WASI_ERRNO_OWNERDEAD
+        | wasi::__WASI_ERRNO_PROTO
+        | wasi::__WASI_ERRNO_RANGE
+        | wasi::__WASI_ERRNO_ROFS
+        | wasi::__WASI_ERRNO_SPIPE
+        | wasi::__WASI_ERRNO_SRCH
+        | wasi::__WASI_ERRNO_TXTBSY
+        | wasi::__WASI_ERRNO_XDEV
+        | wasi::__WASI_ERRNO_NOTCAPABLE => {
+            return std::io::Error::new(std::io::ErrorKind::Other, error_str(errno))
+        }
+        _ => panic!("unrecognized WASI errno value"),
+    } as i32;
+
+    std::io::Error::from_raw_os_error(raw_os_error)
+}
+
+#[cfg(windows)]
+fn error_str(errno: wasi::__wasi_errno_t) -> &'static str {
```

Comment:
Hmm, I guess I must be missing something here but why do we need that here at all? Shouldn't we be able to generate this (as was before as far as I remember) directly from `*.witx`?

Reply:
Ah ok, if you'd like to dig in here (which sounds good to me!) I'll go ahead and keep refactoring this, and with some tweaks of error types this was easily removable.